x-pack/plugins/cases/server/authorization/authorization.ts

```diff
@@ -74,66 +73,45 @@ export class Authorization {
       caseClasses = new Set<string>();
     }
 
-    return new Authorization({ request, securityAuth, caseClasses, isAuthEnabled });
+    return new Authorization({ request, securityAuth, caseClasses, auditLogger });
   }
 
   private shouldCheckAuthorization(): boolean {
     return this.securityAuth?.mode?.useRbacForRequest(this.request) ?? false;
   }
 
-  public async ensureAuthorized(className: string, operation: ReadOperations | WriteOperations) {
-    // TODO: remove
-    if (!this.isAuthEnabled) {
-      return;
-    }
-
+  public async ensureAuthorized(className: string, operation: OperationDetails) {
     const { securityAuth } = this;
     const isAvailableClass = this.featureCaseClasses.has(className);
 
-    // TODO: throw if the request is not authorized
     if (securityAuth && this.shouldCheckAuthorization()) {
-      // TODO: implement ensure logic
-      const requiredPrivileges: string[] = [securityAuth.actions.cases.get(className, operation)];
+      const requiredPrivileges: string[] = [
+        securityAuth.actions.cases.get(className, operation.name),
+      ];
 
       const checkPrivileges = securityAuth.checkPrivilegesDynamicallyWithRequest(this.request);
-      const { hasAllRequested, username, privileges } = await checkPrivileges({
+      const { hasAllRequested, username } = await checkPrivileges({
         kibana: requiredPrivileges,
       });
 
       if (!isAvailableClass) {
-        // TODO: throw if any of the class are not available
         /**
          * Under most circumstances this would have been caught by `checkPrivileges` as
          * a user can't have Privileges to an unknown class, but super users
          * don't actually get "privilege checked" so the made up class *will* return
          * as Privileged.
          * This check will ensure we don't accidentally let these through
          */
-        // TODO: audit log using `username`
-        throw Boom.forbidden('User does not have permissions for this class');
+        throw Boom.forbidden(this.auditLogger.failure({ username, className, operation }));
       }
 
       if (hasAllRequested) {
-        // TODO: user authorized. log success
+        this.auditLogger.success({ username, operation, className });
       } else {
-        const authorizedPrivileges = privileges.kibana.reduce<string[]>((acc, privilege) => {
-          if (privilege.authorized) {
-            return [...acc, privilege.privilege];
-          }
-          return acc;
-        }, []);
-
-        const unauthorizedPrivilages = requiredPrivileges.filter(
-          (privilege) => !authorizedPrivileges.includes(privilege)
```

Comment:
I looked through the alerting code and it seems like they're only using these to determine the `consumer` and `scope`. We don't have a concept of those and it didn't look like they were actually logging the either of the authorized or unauthorized privileges in the message so I don't think we need these 🤷‍♂️  let me know what you think @cnasikas

Reply:
Make sense to me!